### PR TITLE
GUI: Show link to the page that allow to add a new token

### PR DESCRIPTION
**Pull request by ogoffart**
If owncloud >= 9.1 is detexted:
instead of "Password:" ask for "Password or Token"
and add a link to the ownCloud page that allow to add device token

Issue #4877

### CODE REVIEW

**Comment by ogoffart** [comment]
cc @danimo @dragotin @ChristophWurst 

Note that the ownCloud server's GUI says "device password" and i'm using "device token"  

Maybe there is no need for https://github.com/owncloud/core/issues/25091 after all.

**Comment by guruz** [comment]
> Note that the ownCloud server's GUI says "device password" and i'm using "device token"

Euhm.. can you guys agree to a common naming? 
I'd even say "application token" might make more sense

@PVince81

**Comment by PVince81** [comment]
Yeah, maybe let's also call it "Device password" or "Password" in the client.
The link could be called "Create device specific password"

**Comment by ogoffart** [comment]
After discussion with @ChristophWurst,  we decided to rename it "App Password"

**Comment by ogoffart** [comment]
The difference now is that there is a message: "Click here to request an app password from the browser."

File: src/gui/wizard/owncloudhttpcredspage.cpp [hunk 1]
```diff
@@ -114,6 +114,16 @@ void OwncloudHttpCredsPage::initializePage()
             _ui.lePassword->setText(password);
         }
     }
+    bool hasTokenAuth = ocWizard->account()->serverVersionInt() >= 0x090100;
+    if (hasTokenAuth) {
+        _ui.tokenLabel->setText(tr("<a href=\"%1/index.php/settings/personal#apppasswords\">Click here</a>"
+                        " to requestn app password from the browser.")
```
Comment (guruz) [inline src/gui/wizard/owncloudhttpcredspage.cpp]: "requestn"

File: src/gui/creds/httpcredentialsgui.cpp [hunk 2]
```diff
@@ -31,22 +32,38 @@ void HttpCredentialsGui::askFromUser()
...
+                     "User: %2<br>"
+                     "Account: %3<br>")
+                  .arg(Utility::escape(Theme::instance()->appNameGUI()),
+                       Utility::escape(_user),
+                       Utility::escape(_account->displayName()));
+
+    bool hasTokenAuth = _account->serverVersionInt() >= 0x090100;
+    if (hasTokenAuth) {
+        msg += tr("<br><a href=\"%1/index.php/settings/personal#apppasswords\">Click here</a>"
+                     " to request an app password from the browser.<br>")
```
Comment (guruz) [inline src/gui/creds/httpcredentialsgui.cpp]: s/browser/web interface/ ?

File: src/gui/wizard/owncloudhttpcredspage.cpp (hunk 1 above)
Comment (guruz) [inline src/gui/wizard/owncloudhttpcredspage.cpp]: s/browser/web interface/ ?

make sure both strings are the same for i18n?

**Comment by guruz** [comment]
👍

**Comment by dragotin** [comment]
:+1: nice

**Comment by PVince81** [comment]
Requires core's https://github.com/owncloud/core/pull/25134

**Comment by guruz** [comment]
Change URL pls
https://github.com/owncloud/core/pull/25134#issuecomment-227462405

**Comment by ogoffart** [comment]
Updated with the new URL

**Comment by guruz** [comment]
👍

**Pull request merged**